pootle/static/js/shared/components/AutoIcon.js

```diff
@@ -40,81 +38,57 @@ function calcHue(title) {
   return crc16(title) % 360;
 }
 
-class AutoIcon extends React.Component {
-
-  static propTypes() {
-    return {
-      title: React.PropTypes.string.isRequired,
-      saturation: React.PropTypes.number,
-      lightness: React.PropTypes.number,
-      abbreviationLength: React.PropTypes.number,
-      mode: React.PropTypes.string,
-      size: React.PropTypes.number,
-    };
-  }
-
-  constructor(props) {
-    super();
-
-    this.state = assign({
-      saturation: 100,
-      lightness: 30,
-      abbreviationLength: 2,
-      mode: 'solid',
-      size: 20,
-    }, props);
-  }
-
-  componentDidMount() {
-    this.calculateProps(this.props.title);
-  }
-
-  componentWillReceiveProps(newProps) {
-    if (this.props.title !== newProps.title) {
-      this.calculateProps(newProps.title);
-    }
-  }
-
-  calculateProps(title) {
-    this.setState({
-      abbreviation: calcAbbreviation(title, this.state.abbreviationLength),
-      hue: calcHue(title),
-    });
-  }
-
-  render() {
-    const color =
-      `hsl(${this.state.hue}, ${this.state.saturation}%, ${this.state.lightness}%)`;
-
-    const size = `${this.state.size}px`;
-    const fontSize = `${Math.round(this.state.size / 2)}px`;
-
-    const style = (this.state.mode === 'solid') ? {
-      backgroundColor: color,
-      border: `1px solid ${color}`,
-      color: '#fff',
-      width: size,
-      height: size,
-      lineHeight: size,
-      fontSize,
-    } : {
-      border: `1px solid ${color}`,
-      color,
-      width: size,
-      height: size,
-      lineHeight: size,
-      fontSize,
-    };
-
-    return (
-      <span
-        className="auto-icon"
-        style={style}
-      >{this.state.abbreviation}</span>
-    );
-  }
-
+function AutoIcon(props) {
+  const abbreviation = calcAbbreviation(props.title, props.abbreviationLength);
+  const hue = calcHue(props.title);
+
+  const color =
+    `hsl(${hue}, ${props.saturation}%, ${props.lightness}%)`;
+
+  const size = `${props.size}px`;
+  const fontSize = `${Math.round(props.size / 2)}px`;
+
+  const style = (props.mode === 'solid') ? {
```

Comment:
Not related to the PR but I realize this can be made more compact, as the geometry always matches:
```js
const style = {
  border: `1px solid ${color}`,
  width: size,
  height: size,
  lineHeight: size,
  color,
  fontSize,
};
if (mode === 'solid') {
  style.backgroundColor = color;
  style.color = '#fff';
}
```